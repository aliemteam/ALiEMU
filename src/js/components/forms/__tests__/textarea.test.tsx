import React from 'react';
import renderer from 'react-test-renderer';

import Textarea from '../textarea';

it('should render correctly', () => {
    const tree = renderer.create(
        <>
            <Textarea />
            <Textarea label="Labelled Textarea" />
            <Textarea disabled label="Labelled Disabled Textarea" />
            <Textarea defaultValue="12345" maxLength={10} />
            <Textarea
                label="Labelled with maxlength"
                defaultValue="12345"
                maxLength={10}
            />
        </>,
    );
    expect(tree).toMatchSnapshot();
});
