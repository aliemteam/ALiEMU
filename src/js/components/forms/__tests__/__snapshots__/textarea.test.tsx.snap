// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <textarea
    className="textarea"
    onChange={[Function]}
    style={Object {}}
  />,
  <label
    className=""
  >
    Labelled Textarea
    <textarea
      className="textarea"
      onChange={[Function]}
      style={Object {}}
    />
  </label>,
  <label
    className="undefined"
  >
    Labelled Disabled Textarea
    <textarea
      className="textarea"
      disabled={true}
      onChange={[Function]}
      style={Object {}}
    />
  </label>,
  <div
    className="progressRow"
  >
    <textarea
      className="textarea"
      defaultValue="12345"
      maxLength={10}
      onChange={[Function]}
      style={Object {}}
    />
    <div
      aria-valuemax={10}
      aria-valuemin={0}
      aria-valuenow={5}
      role="progressbar"
    >
      <div
        style={
          Object {
            "width": "50%",
          }
        }
      />
    </div>
  </div>,
  <label
    className=""
  >
    Labelled with maxlength
    <div
      className="progressRow"
    >
      <textarea
        className="textarea"
        defaultValue="12345"
        maxLength={10}
        onChange={[Function]}
        style={Object {}}
      />
      <div
        aria-valuemax={10}
        aria-valuemin={0}
        aria-valuenow={5}
        role="progressbar"
      >
        <div
          style={
            Object {
              "width": "50%",
            }
          }
        />
      </div>
    </div>
  </label>,
]
`;
